tests/components/demo/test_media_player.py

```diff
@@ -60,6 +60,27 @@ async def test_source_select(hass):
     assert state.attributes.get(mp.ATTR_INPUT_SOURCE) == "xbox"
 
 
+async def test_repeat_set(hass):
+    """Test the repeat set service."""
+    entity_id = "media_player.walkman"
+
+    assert await async_setup_component(
+        hass, mp.DOMAIN, {"media_player": {"platform": "demo"}}
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get(entity_id)
+    assert state.attributes.get(mp.ATTR_MEDIA_REPEAT) == mp.const.REPEAT_MODE_OFF
+
+    await hass.services.async_call(
+        mp.DOMAIN,
+        mp.SERVICE_REPEAT_SET,
+        {ATTR_ENTITY_ID: entity_id, mp.ATTR_MEDIA_REPEAT: mp.const.REPEAT_MODE_ALL},
+        blocking=True,
+    )
+    state = hass.states.get(entity_id)
+    assert state.attributes.get(mp.ATTR_MEDIA_REPEAT) == mp.const.REPEAT_MODE_ALL
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)